AutoFlow_OmicsDataHandling/INCA_script_generator.py

```diff
@@ -0,0 +1,1232 @@
+    mat_script = "m = model(r); % set up model\n\n"
+
+    return mat_script
+
+
+def symmetrical_metabolites(atomMappingMetabolite_data_I):
+    """
+    Takes care of symmetrical metabolites if not done so in the
+    reaction equations
+
+    Parameters:
+        atomMappingMetabolite_data_I: pre-processed
+            atomMappingMetabolite_data_I input data
+
+    Returns:
+        mat_script: addition to MATLAB script under construction
+
+    """
+    tmp_script = "% take care of symmetrical metabolites\n"
+    for cnt_met, met in atomMappingMetabolite_data_I.iterrows():
+        if not pd.isna(met["met_symmetry_atompositions"]):
+            tmp_script = (
+                tmp_script
+                + "m.mets{'"
+                + met["met_id"]
+                + "'}.sym = list('rotate180', atommap('"
+            )
+            met_positions = [
+                int(x)
+                for x in prepare_input(met["met_atompositions"], "Curly")
+            ]
+            met_elements = prepare_input(met["met_elements"], "Curly")
+            met["met_symmetry_elements"] = met[
+                "met_symmetry_elements"
+            ].replace("{}", str(np.nan))
+            met_symmetry_elements = prepare_input(
+                met["met_symmetry_elements"], "Curly"
+            )
+            met["met_symmetry_atompositions"] = met[
+                "met_symmetry_atompositions"
+            ].replace("{}", str(np.nan))
+            met_symmetry_atompositions = [
+                int(x)
+                for x in prepare_input(
+                    met["met_symmetry_atompositions"], "Curly"
+                )
+            ]
+
+            for cnt, atompositions in enumerate(met_positions):
+                tmp_script = (
+                    tmp_script
+                    + met_elements[cnt]
+                    + str(atompositions + 1)
+                    + ":"
+                    + met_symmetry_elements[cnt]
+                    + str(met_symmetry_atompositions[cnt] + 1)
+                    + " "
+                )
+            tmp_script = tmp_script[:-1]
+            tmp_script = tmp_script + "'));\n"
+        else:
+            continue
+    tmp_script = tmp_script + "\n"
+    mat_script = tmp_script
+
+    return mat_script
+
+
+def unbalanced_reactions(atomMappingMetabolite_data_I):
+    # NOTE: hard-coded for now until a better workaround can be done
+
+    """
+    Adds in the metabolite state (balanced or unbalanced)
+
+    Parameters:
+        atomMappingMetabolite_data_I: pre-processed
+            atomMappingMetabolite_data_I input data
+
+    Returns:
+        mat_script: addition to MATLAB script under construction
+
+    """
+    tmp_script = "% define unbalanced reactions\n"
+    # specify reactions that should be forcible unbalanced
+    metabolites_all = [
+        x["met_id"] for cnt, x in atomMappingMetabolite_data_I.iterrows()
+    ]
+    # unbalanced reactions:
+    for met in ["co2_e", "h2o_e", "h_e", "na1_e"]:
```

Comment:
I would recommend moving this list to a default parameter so the user has a bit more control over the "hard coding"